src/pip/_internal/commands/__init__.py

```diff
@@ -101,6 +98,7 @@ def create_command(name, **kwargs):
 
 
 def get_similar_commands(name):
+    # type: (str) -> Optional[Sequence[str]]
```

Comment:
Again, the return type is `Optional[str]`.  Because of this reason, could you please rename the function accordingly (singular *command*)?

Reply:
Hey @McSinyx , 

The return type is `Optional[Sequence[str]]` since the return type of 
`difflib.get_close_matches` is ```builtins.list[typing.Sequence[_T-1]]```  which equates to `Optional[Sequence[str]]` as you can observe below from the output of `reveal_type`, hence the return type of `get_similar_commands` which takes the first element of this return value is `[Sequence[str]]`, or `Optional[Sequence[str]]` in our case, since we return too as well.

```
from difflib import get_close_matches
reveal_type(get_close_matches)
#  Revealed type is 'def [_T] (word: typing.Sequence[_T`-1], 
# possibilities: typing.Iterable[typing.Sequence[_T`-1]], 
# n: builtins.int =, cutoff: builtins.float =) 
# -> builtins.list[typing.Sequence[_T`-1]]'
```

Reply:
Yep, but that's not what's returned: https://github.com/pypa/pip/pull/7977/files#diff-25a31740fea4e0eefa38ee5b171d9c3dR109-R112

Reply:
Also, since you marks the outdated one as resolved, it should be considered if callers of this function should explicitly check for `None`.

Reply:
> Yep, but that's not what's returned: https://github.com/pypa/pip/pull/7977/files#diff-25a31740fea4e0eefa38ee5b171d9c3dR109-R112

Sorry I didn't get what you were pointing to.  Since we have `return close_commands[0]` which returns a `Sequence[str]` and `return None`, I am in finally returning `Optional[Sequence[str]]`, is there something wrong with that?

Reply:
> Also, since you marks the outdated one as resolved, it should be considered if callers of this function should explicitly check for `None`.

I think that the `if guess:` checks at https://github.com/pypa/pip/blob/master/src/pip/_internal/commands/help.py#L33 and https://github.com/pypa/pip/blob/master/src/pip/_internal/cli/main_parser.py#L90 should handle the `None check` right? Since `if guess:` will evaluate to False and a `CommandError` will be raised.

Reply:
> Again, the return type is `Optional[str]`. Because of this reason, could you please rename the function accordingly (singular _command_)?

I just saw your edited comment about the renaming, have fixed that.

Reply:
@deveshks I also think the return type is `Optional[str]`. The rest LGTM.

Reply:
Hi @sbidoul 

When I to use `# type: (str) -> Optional[str]` and run mypy, I get the following error `error: Incompatible return value type (got "Sequence[str]", expected "Optional[str]")`

Also as I outlined above, the return type of `difflib.get_close_matches` as observed from `reveal_type` is `List[Sequence[str]]`, hence the return type of `get_similar_command`, since we return an element of List, which is of type `Sequence[str]`.

Hence I am using `Sequence[str]`. Is there something incorrect in my understanding here?

Reply:
@sbidoul @McSinyx are correct that `Optional[str]` would be the correct return type here. However, we've surfaced a typeshed bug here, so that needs to be fixed first. See my comment in the parallel review comments thread.